### PR TITLE
feat: implement integration with Dungeon Master's Vault

### PR DESCRIPTION
**Pull request by emjaytee404**
Hi! Thank you for your help and encouragement in trying this out. Let me know if you have any questions.

### CODE REVIEW

File: src/dungeonmastersvault.ts [hunk 1]
```diff
@@ -0,0 +1,388 @@
+/** @format */
+
+import createLogger from './log';
+import { getStorage, migrateStorage } from './storage';
+import { IRoll, ThreeDDiceRollEvent, ThreeDDice, ITheme, ThreeDDiceAPI } from 'dddice-js';
+
+import './index.css';
+import './dungeonmastersvault.css';
+import imageLogo from 'url:./assets/dddice-32x32.png';
+
```
Comment (CelesteBloodreign) [inline src/dungeonmastersvault.ts:12]:
```suggestion
import notify from './utils/notify';
```

You need to import my, now confusingly named, notify lib for the error messages

File: src/dungeonmastersvault.ts [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+function translateD10xs(die) {
+  return die === 'd10x' ? 'd100' : die;
+}
+
+function diceSizeCompare(die1, die2) {
+  return translateD10xs(die1).split('d')[1] - translateD10xs(die2).split('d')[1];
+}
+
+function generateNotificationMessage(roll: IRoll) {
+  const diceBreakdown = roll.values
```
Comment (CelesteBloodreign) [inline src/dungeonmastersvault.ts]: This variable is now unused

File: src/dungeonmastersvault.ts [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+    .filter(die => !die.is_dropped)
+    .reduce(
+      (prev, current) =>
+        prev +
+        (prev !== '' && current.value_to_display[0] !== '-' ? '+' : '') +
+        (typeof current.value_to_display === 'object' ? '⚠' : current.value_to_display),
+      '',
+    );
+
+  const largestDie = translateD10xs(
```
Comment (CelesteBloodreign) [inline src/dungeonmastersvault.ts]: This variable is also now unused!

File: src/dungeonmastersvault.ts [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+    case 'preloadTheme':
+      preloadTheme(message.theme);
+  }
+});
+
+window.addEventListener('load', () => init());
+window.addEventListener('resize', () => init());
+
+// Subscribe to any DOM mutations and re-run init. We have to
+// observe the app div beause the tables don't have IDs.
```
Comment (CelesteBloodreign) [inline src/dungeonmastersvault.ts]: super nit pick! 

beacse -> because

File: src/dungeonmastersvault.ts [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+        if (room) {
+          dddice.connect(room.slug);
+        }
+        if (theme) {
+          preloadTheme(theme);
+        }
+      } else {
+        log.debug('in render mode off');
+        dddice = new ThreeDDice();
+        dddice.api = new ThreeDDiceAPI(apiKey);
```
Comment (CelesteBloodreign) [inline src/dungeonmastersvault.ts]:
```suggestion
        dddice.api = new ThreeDDiceAPI(apiKey, "Dungeon Master's Vault");
```

File: src/dungeonmastersvault.ts [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+
+    const img = document.createElement('img');
+    img.src = imageLogo;
+    img.className = 'h-auto w-auto';
+
+    const buttonRoll = document.createElement('button');
+    buttonRoll.addEventListener('pointerup', onPointerUp(overlayId));
+    buttonRoll.appendChild(img);
+    buttonRoll.className =
+      'h-8 w-8 bg-gray-900 rounded-l flex items-center justify-center p-1 hover:bg-gray-700 transition-colors duration-80';
```
Comment (CelesteBloodreign) [inline src/dungeonmastersvault.ts]:
```suggestion
      'h-8 w-8 bg-gray-900 rounded-l flex items-center justify-center !p-1 hover:bg-gray-700 transition-colors duration-80';
```

Missed one! :D

**Review by CelesteBloodreign — CHANGES_REQUESTED** [review]
This looks super good! I tested it out and it works well!

I left a few comments that I think need fixing.

Could you also run `npx prettier src --write` to update the code to meet our coding style guidelines automatically!

If it reformats more than just your files, don't check them in, but let me know 👍

File: src/dungeonmastersvault.ts (hunk 1 above)
Comment (emjaytee404) [reply src/dungeonmastersvault.ts:12]: Done! And yeah, it's unfortunate, uh? 😔 I'd love to get rid of the dependency, but my chat messages will look like a five year old designed them. 😂

File: src/dungeonmastersvault.ts (hunk 6 above)
Comment (emjaytee404) [reply src/dungeonmastersvault.ts]: Doh! Fixed!

File: src/dungeonmastersvault.ts (hunk 5 above)
Comment (emjaytee404) [reply src/dungeonmastersvault.ts]: What? I had done this one for sure. I probably lost it in one of the rebases. Sorry.

File: src/dungeonmastersvault.ts (hunk 4 above)
Comment (emjaytee404) [reply src/dungeonmastersvault.ts]: ☹️ j/k, thanks!

File: src/dungeonmastersvault.ts (hunk 2 above)
Comment (emjaytee404) [reply src/dungeonmastersvault.ts]: Ah. Indeed.

File: src/dungeonmastersvault.ts (hunk 3 above)
Comment (emjaytee404) [reply src/dungeonmastersvault.ts]: Taken care of.

**Review by CelesteBloodreign — APPROVED** [review]

**Comment by CelesteBloodreign** [comment]
Looking great! thanks again ❤️

File: src/dungeonmastersvault.ts (hunk 1 above)
Comment (CelesteBloodreign) [reply src/dungeonmastersvault.ts:12]: Maybe we can tackle that after the refactor :D

**Pull request merged**